src/services/items/handler/item-action-handler.ts

```diff
@@ -0,0 +1,142 @@
+import { ActionHandlerInput, BaseAction } from 'graasp-plugin-actions';
+import geoip from 'geoip-lite';
+import { ACTION_TYPES, METHODS, paths, VIEW_UNKNOWN_NAME } from '../constants/constants';
+import { ItemService } from '../db-service';
+import { CLIENT_HOSTS } from '../../../util/config';
+
+export const itemActionHandler = async (
+  dbService: ItemService,
+  actionInput: ActionHandlerInput,
+): Promise<BaseAction[]> => {
+  const { request, log, dbHandler } = actionInput;
+  // function called each time there is a request in the items in graasp (onResponse hook in graasp)
+  // identify and check the correct endpoint of the request
+  // check that the request is ok
+  const { headers, member, method, url, ip, query, params } = request;
+  // warning: this is really dependent on the url -> how to be more safe and dynamic?
+  const paramItemId: string = (params as { id: string })?.id;
+  let queryItemIds = (query as { id })?.id;
+  if (!Array.isArray(queryItemIds)) {
+    queryItemIds = [queryItemIds];
+  }
+  const geolocation = geoip.lookup(ip);
```

Comment:
I see you had to import this plugin just for this line. Do you think we should create util functions imported from the graasp-actions plugin to reuse here? So we keep a maximum information in the action plugin. For example:
- getGeolocationIp
-  getView
- buildBaseAction

The names of the functions are up to you of course.